macros/src/main/scala/org/zalando/grafter/macros/ReaderMacro.scala

```diff
@@ -10,62 +11,52 @@ import ReaderMacros._
  */
 object ReaderMacro {
 
-  val annotation = "reader"
+  val annotationName = "reader"
 
-  def expand(classDef: Defn.Class, objectDef: Option[Defn.Object]): Term.Block = {
-    
-    classDef.ctor.paramss.toList match {
-      case Nil =>
-        output(classDef, objectDef) {
-          val init = template"${Ctor.Ref.Name(classDef.name.value)}"
-          q"""
-              implicit def reader[A]: cats.data.Reader[A, ${classDef.name}] =
-                cats.data.Reader(_ => new $init)
-           """
-        }
-
-      case params :: _ =>
-        val parameters = collectParamTypesAndNamesAsList(params)
+  def impl(c: whitebox.Context)(annottees: c.Expr[Any]*): c.Expr[Any] = {
+    import c.universe._
+    def name(t: Name) = t.decodedName.toString.trim
```

Comment:
Cargo-culting, I took this from somewhere. I hope anyway to get rid of this whole implementation once the new macros are in.